src/mam4xx/coagulation.hpp

```diff
@@ -59,7 +64,1120 @@ class Coagulation {
+//   u.s. environmental protection agency, research triangle park, n.c.,
+//   (ntis pb91-161729/as), 1991
+//
+//   2. binkowski, f.s. an u. shankar, the regional particulate matter
+//   model 1. model decsription and preliminary results, journal of
+//   geophysical research, 100, d12, pp 26,191-26,209,
+//   december 20, 1995.
+//
+//   3. binkowski, f.s. and s.j. roselle, models-3 community
+//   multiscale air quality (cmaq) model aerosol component 1:
+//   model description.  j. geophys. res., vol 108, no d6, 4183
+//   doi:10.1029/2001jd001409, 2003.
+// --------------------------------------------------------
+KOKKOS_INLINE_FUNCTION
+void getcoags(const Real lamda, const Real kfmatac, const Real kfmat,
+              const Real kfmac, const Real knc, const Real dgatk,
+              const Real dgacc, const Real sgatk, const Real sgacc,
+              const Real xxlsgat, const Real xxlsgac, Real &qn11, Real &qn22,
+              Real &qn12, Real &qv12) {
+
+  const Real a_const = 1.246;
+  const Real esat01 = haero::exp(0.125 * xxlsgat * xxlsgat);
+  const Real esac01 = haero::exp(0.125 * xxlsgac * xxlsgac);
+  const Real sqrttwo = haero::sqrt(2.0);
+  const Real dlgsqt2 = 1.0 / haero::log(sqrttwo);
+
+  const Real esat04 = haero::pow(esat01, 4.0);
+  const Real esac04 = haero::pow(esac01, 4.0);
+
+  const Real esat05 = esat04 * esat01;
+  const Real esac05 = esac04 * esac01;
+
+  const Real esat08 = esat04 * esat04;
+  const Real esac08 = esac04 * esac04;
+
+  const Real esat09 = esat08 * esat01;
+  const Real esac09 = esac08 * esac01;
+
+  const Real esat16 = esat08 * esat08;
+  const Real esac16 = esac08 * esac08;
+
+  const Real esat20 = esat16 * esat04;
+  const Real esac20 = esac16 * esac04;
+
+  const Real esat24 = esat20 * esat04;
+
+  const Real esat25 = esat20 * esat05;
+  const Real esac25 = esac20 * esac05;
+
+  const Real esat36 = esat20 * esat16;
+
+  const Real esat49 = esat24 * esat25;
+
+  const Real esat64 = esat20 * esat20 * esat24;
+
+  const Real esat100 = esat64 * esat36;
+
+  const Real dgat3 = dgatk * dgatk * dgatk;
+
+  const Real sqdgat = haero::sqrt(dgatk);
+  const Real sqdgac = haero::sqrt(dgacc);
+  const Real sqdgat7 = dgat3 * sqdgat;
+
+  const Real r1 = sqdgac / sqdgat;
+  const Real r2 = r1 * r1;
+  const Real rx4 = r2 * r2;
+  const Real ri1 = 1.0 / r1;
+  const Real ri2 = 1.0 / (r1 * r1);
+  const Real ri3 = 1.0 / (r1 * r1 * r1);
+  const Real kngat = 2.0 * lamda / dgatk;
+  const Real kngac = 2.0 * lamda / dgacc;
+
+  //  Calculate ratio of geometric mean diameters
+
+  const Real rat = dgacc / dgatk;
+
+  // Trap subscripts for bm0 and bm0i, between 1 and 10.
+  // See page h.5 of whitby et al. (1991)
+  const int n2n =
+      haero::max(1, haero::min(10, std::round(4.0 * (sgatk - 0.75)))) - 1;
+  const int n2a =
+      haero::max(1, haero::min(10, std::round(4.0 * (sgacc - 0.75)))) - 1;
+  const int n1 =
+      haero::max(1, haero::min(10, 1 + std::round(dlgsqt2 * haero::log(rat)))) -
```

Comment:
Looks like Cuda has a round() function.  Probably need to add this to hero's math.hpp to avoid calling std:: on device.

Reply:
Added round to haero now use it here in coagulation.hpp.